sklearn/metrics/_ranking.py

```diff
@@ -696,8 +697,14 @@ def _binary_clf_curve(y_true, y_score, pos_label=None, sample_weight=None):
     assert_all_finite(y_true)
     assert_all_finite(y_score)
 
+    # Filter out zero-weighted samples, as they should not impact the result
     if sample_weight is not None:
         sample_weight = column_or_1d(sample_weight)
+        _check_sample_weight(sample_weight, y_true)
```

Comment:
```suggestion
    sample_weight = _check_sample_weight(sample_weight, y_true)
```

Reply:
`_check_sample_weight` internally calls `check_array`. This proposed change is the only one I consider important and not cosmetic.

Reply:
Done. Thank you for pointing this out.

Reply:
I think you can remove the line `sample_weight = column_or_1d(sample_weight)`.

Reply:
After that change, I think it'll be a LGTM from my side:smirk:

Reply:
Both `y_true` and `y_score` can be given as column vectors, both being raveled by `column_or_1d`. If `sample_weight` is also given as a column vector, it will also need being raveled by `column_or_1d`. Otherwise, `_check_sample_weight` will raise a ValueError: Sample weights must be 1D array or scalar.

Reply:
If `weight.shape = (n_samples, 1)`, `_check_samples_weight` will throw `ValueError: Sample weights must be 1D array or scalar`. That is expected behavior, the doc says: `sample_weight : array-like of shape (n_samples,)`
So I still think we should remove the `column_or_1d`. Am I missing something?

Reply:
All three `y_true`, `y_score` and `sample_weight` should be treated equally. Either all can be given as column vectors, or none. But it makes no sense accepting 2 as column vectors and not the other one. Therefore, either all three pass through `column_or_1d` or none of them, but it makes no sense removing `column_or_1d` only for `sample_weight`.

Reply:
In this case, we can't call `_check_sample_weight`.

Reply:
@scikit-learn/core-devs Could someone help out?
According to the docstrings, `_binary_clf_curve` accepts only 1d array-likes of shape `(n_samples,)`. Nevertheless, the code for `_binary_clf_curve`, dating back more than 7 yeary before #3442, calls `column_or_1d` on `y_true`, `y_score` and `sample_weight` and therefore also accepts shape `(n_samples, 1)`.

Shall we adopt the docstings or be stricter to the inputs and actually require shape `(n_sampels,)`.

Reply:
I don't think we should change the behavior. When there's a discrepancy between the code and the docs, we tend to fix the docs. In this case, I think I'd be happy with having `column_or_1d`, since it makes it more convenient for people to work with the API, and it doesn't have much of a consequence. In terms of the docs, I'm not sure if we need to updated them, but I wouldn't object an update I think. I'd need to think more about it.

Reply:
@adrinjalali Thanks for your expert judgement. In this case, I'd leave the docs as is.
The only open question is then whether to use `_check_sample_weight`, which enforces shape `(n_samples,)`, **xor** `column_or_1d` for `sample_weight`&mdash;but not both.
@albertvillanova For me, the choice is yours:smirk:

Reply:
Oh, I realize: the current solution with both `column_or_1d` before `_check_sample_weight` works as expected.